server/app/filters/SessionTimeoutFilter.java

```diff
@@ -0,0 +1,141 @@
+package filters;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import auth.CiviFormProfile;
+import auth.ProfileUtils;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.nio.charset.StandardCharsets;
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Base64;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+import javax.inject.Inject;
+import javax.inject.Provider;
+import org.apache.pekko.stream.Materializer;
+import play.libs.Json;
+import play.mvc.Filter;
+import play.mvc.Http;
+import play.mvc.Result;
+import services.settings.SettingsManifest;
+
+public class SessionTimeoutFilter extends Filter {
+  public static final int DEFAULT_INACTIVITY_TIMEOUT_MINUTES = 30;
+  public static final int DEFAULT_MAX_SESSION_DURATION_MINUTES = 600;
```

Comment:
```suggestion

```
(blocking) This came up in the previous PR, and I'm going to bring it up again since these are starting to get used elsewhere now.

Just delete these.

Instead of  `settingsManifest.get().getMaximumSessionDurationMinutes().orElse(DEFAULT_MAX_SESSION_DURATION_MINUTES)`

Do `settingsManifest.get().getMaximumSessionDurationMinutes().orElseThrow()`

The default value is already in the config and this is just adding unnecessary code.

Reply:
(blocking) Please remove these and just use the defaults values from the application.conf